docs/storage-surface.rst

```diff
@@ -0,0 +1,673 @@
+
+  - ``project`` (derived from GAE / GCE environ if present).
+
+  - ``bucket`` (derived from GAE / GCE environ if present).
+
+- By calling methods which require authentication
+
+  .. code-block:: python
+
+   >>> from gcloud import storage
+   >>> bucket = storage.get_bucket('my-bucket')
+
+  the default connection "just works" out of the box and will be lazily
+  loaded and stored after the first use.
+
+- Set defaults in a declarative fashion
+
+  .. code-block:: python
+
+   >>> storage.set_defaults()
+
+  or instead of using implicit behavior, pass in defaults
+
+  .. code-block:: python
+
+   >>> storage.set_defaults(connection=connection, project='some-project')
+   ...                      bucket=some_bucket)
+
+  though not all are needed
+
+  .. code-block:: python
+
+   >>> storage.set_defaults(project='some-project', bucket=some_bucket)
+
+- Declaratively set defaults one-by-one
+
+  .. code-block:: python
+
+   >>> storage.set_default_bucket(some_bucket)
+   >>> storage.set_default_connection(connection)
+   >>> storage.set_default_project(project)
+
+
+Manage buckets
+--------------
+
+Create a new bucket
+
+.. code-block:: python
+
+   >>> from gcloud import storage
+   >>> new_bucket = storage.create_bucket(bucket_name)
+
+if you desire to be declarative, you may pass in a connection to
+override the default
+
+.. code-block:: python
+
+   >>> new_bucket = storage.create_bucket(bucket_name, connection=connection)
+
+.. note::
+  All methods in the ``storage`` package accept ``connection`` as an optional
+  parameter.
+
+Retrieve an existing bucket
+
+.. code-block:: python
+
+   >>> existing_bucket = storage.get_bucket(bucket_name)
+
+but if the bucket does not exist an exception will occur
+
+.. code-block:: python
+
+   >>> existing_bucket = storage.get_bucket(bucket_name)
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   gcloud.exceptions.NotFound: 404 Some Message
+
+To get a null response instead of an exception, use
+:func:`lookup_bucket <gcloud.storage.api.lookup_bucket>`
+
+.. code-block:: python
+
+   >>> non_existent = storage.lookup_bucket(bucket_name)
+   >>> print non_existent
+   None
+
+To retrieve multiple buckets in a single request
+
+.. code-block:: python
+
+   >>> bucket1, bucket2, bucket3 = storage.get_buckets('bucket-name1',
+   ...                                                 'bucket-name2',
+   ...                                                 'bucket-name3')
+
+This is equivalent to
+
+.. code-block:: python
+
+   >>> with storage.Batch():
+   ...     bucket1 = storage.get_bucket('bucket-name1')
+   ...     bucket2 = storage.get_bucket('bucket-name2')
+   ...     bucket3 = storage.get_bucket('bucket-name3')
+
+To list all buckets associated to the default project
+
+.. code-block:: python
+
+   >>> for bucket in storage.list_buckets():
+   ...     print bucket
+   <Bucket: foo>
+   <Bucket: bar>
+   <Bucket: baz>
```

Comment:
Yes. Wrapping up the concept of an iterator and list into the same method. `get_all` is actually a really bad idea (at least for blobs), so this encourages viewing it just as a way to make queries.